cwe/cwe.go

```diff
@@ -0,0 +1,111 @@
+package cwe
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/aquasecurity/vuln-list-update/utils"
+	"golang.org/x/xerrors"
+)
+
+type CWEConfig struct {
+	url        string
+	retryTimes int
+	cweDir     string
+}
+
+func NewCWEConfig() CWEConfig {
+	return NewCWEWithConfig("https://cwe.mitre.org/data/xml/cwec_latest.xml.zip", filepath.Join(utils.VulnListDir(), "cwe"), 5)
+}
+
+func NewCWEWithConfig(url, path string, retryTimes int) CWEConfig {
+	return CWEConfig{
+		url:        url,
+		retryTimes: retryTimes,
+		cweDir:     path,
+	}
+}
+
+func (c CWEConfig) Update() error {
+	log.Println("Fetching CWE data...")
+	data, err := utils.FetchURL(c.url, "", c.retryTimes)
+	if err != nil {
+		return xerrors.Errorf("failed to fetch cwe data: %w", err)
+	}
+
+	b, err := c.unzip(data)
+	if err != nil {
+		return err
+	}
+
+	var wc WeaknessCatalog
+	if wc, err = xmlToJSON(b); err != nil {
+		return err
+	}
+
+	for _, w := range wc.Weaknesses.Weakness {
+		b, err := json.MarshalIndent(w, "", " ")
+		if err != nil {
+			log.Printf("unable to marshal: %d, err: %s\n", w.ID, err)
+			continue
+		}
+		if err := c.saveFile(b, fmt.Sprintf("CWE-%d.json", w.ID)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c CWEConfig) saveFile(b []byte, fileType string) error {
+	if err := os.MkdirAll(c.cweDir, os.ModePerm); err != nil {
+		return xerrors.Errorf("unable to create cwe directory: %w", err)
+	}
```

Comment:
nit: I think we don't need to try creating the directory every time. I mean we can define this process outside of `saveFile` and the iteration of `wc.Weakness.Weakness`.

Reply:
good point! I've fixed that.